javalib/src/main/scala/java/util/AbstractList.scala

```diff
@@ -1,3 +1,284 @@
 package java.util
 
-abstract class AbstractList[E] extends AbstractCollection[E] with List[E]
+import scala.collection.JavaConversions._
+
+import scala.annotation.tailrec
+
+abstract class AbstractList[E] protected () extends AbstractCollection[E] with List[E] {
+  self =>
+
+  override def add(element: E): Boolean = {
+    add(size, element)
+    true
+  }
+
+  def set(index: Int, element: E): E =
+    throw new UnsupportedOperationException
+
+  def add(index: Int, element: E): Unit =
+    throw new UnsupportedOperationException
+
+  def remove(index: Int): E =
+    throw new UnsupportedOperationException
+
+  def indexOf(o: Any): Int =
+    iterator().indexWhere(_ === o)
+
+  def lastIndexOf(o: Any): Int = {
+    @tailrec
+    def findIndex(iter: ListIterator[E]): Int = {
+      if (!iter.hasPrevious) -1
+      else if (iter.previous() === o) iter.nextIndex
+      else findIndex(iter)
+    }
+    findIndex(listIterator(size))
+  }
+
+  override def clear(): Unit =
+    removeRange(0, size)
+
+  def addAll(index: Int, c: Collection[_ <: E]): Boolean = {
+    checkIndexOnBounds(index)
+    for ((elem, i) <- c.iterator().zipWithIndex)
+      add(index + i, elem)
+    c.nonEmpty
+  }
+
+  def iterator(): Iterator[E] =
+    listIterator()
+
+  def listIterator(): ListIterator[E] =
+    listIterator(0)
+
+  def listIterator(index: Int): ListIterator[E] = {
+    checkIndexOnBounds(index)
+    new RandomAccessListIterator(index, 0, size)
```

Comment:
This needs a comment why it always return a `RandomAccessListInterator`, instead of of a `BackedUpListIterator`.